docs/adr/ADR-005-User_attribution_immutable_state_records.md

```diff
@@ -0,0 +1,197 @@
+
+### Alternatives considered
+
+#### 1. Add user_id to all models (mutable and immutable)
+
+Store a `user_id` foreign key on every model in the application, including mutable records like `Participant`, `Clinic`, etc.
+
+**Pros:**
+
+- Simple, consistent pattern across all models
+- Easy to implement - just add the field to the base model
+- Always know who last touched any record
+
+**Cons:**
+
+- **Loses historical context**: For mutable records, you only see the last user who modified it, not who made which changes
+- **Misleading information**: A `user_id` on a `Participant` that was updated 5 times by different users gives incomplete information
+- **Violates YAGNI**: We have no current functional requirement for attribution on mutable records
+- **Forces premature decision**: Commits us to a specific attribution pattern before we understand the requirements
+- **Audit confusion**: Might create confusion about the relationship between functional attribution and audit logs
+
+#### 2. Use audit logs for functional purposes
+
+Query the `AuditLog` table to determine who performed actions, rather than storing user attribution directly on functional models.
+
+**Pros:**
+
+- No additional fields needed on models
+- Reuses existing audit infrastructure
+- Complete historical context is available
+
+**Cons:**
+
+- **Violates ADR-004 principle**: ADR-004 explicitly decided that audit logs should not be used for functional purposes
+- **Coupling risk**: Application functionality would break if audit logs are archived, pruned, or moved to cold storage
+- **Performance concerns**: Querying audit logs for functional UI operations adds unnecessary overhead
+- **Separation of concerns**: Mixes security/compliance data with functional application data
+- **Different lifecycle**: Audit data may have different retention policies than functional data
+
+#### 3. Create separate attribution tables
+
+Create dedicated attribution tables (e.g., `AppointmentStatusAttribution`) that track user actions separate from both the main models and audit logs.
+
+**Pros:**
+
+- Clear separation between functional data, attribution, and audit
+- Flexible schema for attribution-specific fields (reason, notes, etc.)
+- Could support multiple attribution records per state (e.g., "created by" and "approved by")
+
+**Cons:**
+
+- **Significant over-engineering**: Creates extra complexity for a simple need
+- **Additional joins**: Requires extra database joins for every query that needs attribution
+- **More tables to maintain**: Additional models, migrations, and schema complexity
+- **Violates YAGNI**: We don't have requirements for multiple attributions, approval chains, or attribution-specific metadata
+- **Harder to query**: Simple queries like "show me this status and who created it" require joins
+
+#### 4. Only add user_id to immutable state records (chosen approach)
```

Comment:
Adding user id to immutable state records seems like a no-brainer, but I'm less sure about the part that says we **never add it to mutable state records**.

I'm wondering why we need to rule that out now... could we instead focus on immutable state records now and then revisit attribution of mutable state if and when we need it?